### PR TITLE
Replace Export privileges group dialog with a modal

### PR DESCRIPTION
**Pull request by shucon**
### Description

Replaces the export privileges group jQuery UI dialog with a Bootstrap modal.

Screenshot from 2021-05-28 22-18-58

Before submitting pull request, please review the following checklist:

- [ ] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [ ] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

**Review by williamdes — APPROVED** [review]
LGTM

File: js/src/server/privileges.js [hunk 1]
```diff
@@ -31,6 +31,29 @@ function checkAddUser (theForm) {
...
 }
 
+/**
+ * Export privileges modal handler
+ *
+ * @param data
+ *
+ * @param modal
+ *
+ * @param $msgbox
```
Comment (williamdes) [inline js/src/server/privileges.js:41]: without types they are not very usefull ;)

Comment (shucon) [reply js/src/server/privileges.js:41]: Yeah, should I remove these comments?

Comment (williamdes) [reply js/src/server/privileges.js:41]: oh no no, just add types for them ;)

File: js/src/server/privileges.js [hunk 2]
```diff
@@ -31,6 +31,29 @@ function checkAddUser (theForm) {
...
+    if (typeof data !== 'undefined' && data.success === true) {
+        modal.find('.modal-body').first().html(data.message);
+        $('#exportPrivilegesModalLabel').first().html(data.title);
+        Functions.ajaxRemoveMessage(msgbox);
+        // Attach syntax highlighted editor to export dialog
+        Functions.getSqlEditor(modal.find('textarea'));
+    } else {
+        modal.modal('hide');
+        Functions.ajaxShowMessage(data.error, false);
+    }
```
Comment (williamdes) [inline js/src/server/privileges.js:54]:
```suggestion
    if (typeof data !== 'undefined' && data.success === true) {
        modal.find('.modal-body').first().html(data.message);
        $('#exportPrivilegesModalLabel').first().html(data.title);
        Functions.ajaxRemoveMessage(msgbox);
        // Attach syntax highlighted editor to export dialog
        Functions.getSqlEditor(modal.find('textarea'));
        return;
    }
    modal.modal('hide');
    Functions.ajaxShowMessage(data.error, false);
```

That will avoid the useless else

**Review by williamdes — APPROVED** [review]

File: libraries/classes/Controllers/Server/PrivilegesController.php [hunk 3]
```diff
@@ -415,6 +415,7 @@ public function index(): void
                     'checkprivsdb' => $_GET['checkprivsdb'],
                     'checkprivstable' => $_GET['checkprivstable'],
                 ]));
+                $this->response->addHTML($tableController->getExportModal());
```
Comment (mauriciofauth) [inline libraries/classes/Controllers/Server/PrivilegesController.php:418]:
```suggestion
                $this->render('export_modal');
```

File: libraries/classes/Controllers/Server/PrivilegesController.php [hunk 4]
```diff
@@ -424,6 +425,7 @@ public function index(): void
                 $this->response->addHTML($databaseController->index([
                     'checkprivsdb' => $_GET['checkprivsdb'],
                 ]));
+                $this->response->addHTML($databaseController->getExportModal());
```
Comment (mauriciofauth) [inline libraries/classes/Controllers/Server/PrivilegesController.php:428]:
```suggestion
                $this->render('export_modal');
```

File: libraries/classes/Controllers/Database/PrivilegesController.php [hunk 5]
```diff
@@ -63,4 +63,9 @@ public function index(array $params): string
             'privileges' => $privileges,
         ]);
     }
+
+    public function getExportModal(): string
+    {
+        return $this->template->render('export_modal');
```
Comment (mauriciofauth) [inline libraries/classes/Controllers/Database/PrivilegesController.php:69]: You can render the template directly, there is no need for this method.

File: libraries/classes/Controllers/Table/PrivilegesController.php [hunk 6]
```diff
@@ -68,4 +68,9 @@ public function index(array $params): string
             'privileges' => $privileges,
         ]);
     }
+
+    public function getExportModal(): string
+    {
+        return $this->template->render('export_modal');
```
Comment (mauriciofauth) [inline libraries/classes/Controllers/Table/PrivilegesController.php:74]: You can render the template directly, there is no need for this method.

File: libraries/classes/Server/Privileges.php [hunk 7]
```diff
@@ -2176,6 +2176,16 @@ public function getUsersOverview($result, array $dbRights, $textDir)
...
     }
 
+    /**
+     * Get HTML for export modal
+     *
+     * @return string HTML snippet
+     */
+    public function getExportModal(): string
+    {
+        return $this->template->render('export_modal');
```
Comment (mauriciofauth) [inline libraries/classes/Server/Privileges.php:2186]: You can render the template directly, there is no need for this method.

File: libraries/classes/Server/Privileges.php [hunk 8]
```diff
@@ -3170,6 +3180,7 @@ public function getHtmlForUserOverview($textDir)
                     $dbRights,
                     $textDir
                 );
+                $usersOverview .= $this->getExportModal();
```
Comment (mauriciofauth) [inline libraries/classes/Server/Privileges.php:3183]:
```suggestion
                $usersOverview .= $this->template->render('export_modal');
```

**Review by MauricioFauth — CHANGES_REQUESTED** [review]
You can rename `templates/export_modal.twig` to `templates/server/privileges/export_modal.twig`

**Review by MauricioFauth — CHANGES_REQUESTED** [review]
For some reason, it's not working with the Bootstrap theme.

image

**Comment by shucon** [comment]
> For some reason, it's not working with the Bootstrap theme.
> 
> image

Was happening due to a race condition. Fixed it, thanks.

File: js/src/server/privileges.js [hunk 9]
```diff
@@ -31,6 +31,28 @@ function checkAddUser (theForm) {
...
 
+/**
+ * Export privileges modal handler
+ *
+ * @param {object} data
+ *
+ * @param {object} modal
+ *
+ * @param {JQuery} msgbox
+ *
```
Comment (williamdes) [inline js/src/server/privileges.js:42]: callback is type function :)

Comment (shucon) [reply js/src/server/privileges.js:42]: Yeah, actually it didn't work. I'm seeing what's wrong.

**Review by williamdes — APPROVED** [review]

**Pull request merged**